source/Classes/030.DnsRecordA.ps1

```diff
@@ -58,6 +58,13 @@ class DnsRecordA : DnsRecordBase
             $dnsParameters['ZoneScope'] = $this.ZoneScope
         }
 
+        # Using -Node parameter with Get-DnsServerResourceRecord if dealing with **same as parrent folder** record.
+        if ($this.Name -in '@', '.', $this.ZoneName)
+        {
+            $dnsParameters.Remove('Name')
```

Comment:
Instead of removing 'Name' that was set prior, can we add a else-block that adds Name to the hashtable when it is needed?